common/credsregistry/registry.go

```diff
@@ -1,5 +1,5 @@
 // nolint: ireturn
-package utils
+package credsregistry
```

Comment:
This registry is not just for credentials. It has general purpose we can read any values from any file or env that are not necessarily credentials. Also that is why most of the variables/methods don't really mention much about credentials, secrets, tokens, etc in the package. That is why those are called `Readers` because they read anything. 

There is probably one or two variables, but those are to be modified so that it won't refer to anything auth related. 

It just happens to be the case that Auth is currently the only example we have that is using this package for. Please change revert these changes. 

You are welcome to create a credsregistry package as wrapper package of these readers with an isolated purpose.

Reply:
Makes sense. I wasn't sure about purpose of other methods.

I have cleaned up the code such that:
* Moved registry from `utils` to `scanning` package which is preserving multipurpose reading intent.
* CredentialsRegistry is now `scanning.Registry`
* Readers interface and 3 impls are moved to a sibling file `readers.go`
* Field rename: Readers.CredKey -> Readers.KeyName
* Error: "credential not found" -> "value not found"

As a result `scanning` has no trace of `creds` words. Everything else is preserved.

Reply:
Now wrapper is called `credscanning` implying relationship to `scanning` package.

Important methods:
* NewJSONProviderCredentials
* NewENVProviderCredentials
* ProviderCredentials.Get - utilises MustString
* ProviderCredentials.GetOauthToken - returns AcessToken and optional Refresh
* LoadPath - JSON file path to put creds based on provider name

Maybe when other requirement will arise we could add wrappers under `scanning` directory.

Reply:
All new changes are here: https://github.com/amp-labs/connectors/pull/668/commits/5be7a2267de0e7e06d363089e4e948f5ccdd20d6

Example running Zendesk Support - Read, Write.
![image](https://github.com/user-attachments/assets/0e0b73e8-e47d-4539-bd26-ea37a79b3c52)
![image](https://github.com/user-attachments/assets/003829a4-a534-4cfb-bb0c-eb3144417c97)

Reply:
All new changes are here: https://github.com/amp-labs/connectors/pull/668/commits/5be7a2267de0e7e06d363089e4e948f5ccdd20d6

Example running Zendesk Support - Read, Write.
![image](https://github.com/user-attachments/assets/0e0b73e8-e47d-4539-bd26-ea37a79b3c52)
![image](https://github.com/user-attachments/assets/003829a4-a534-4cfb-bb0c-eb3144417c97)